pkg/reconciler/identity/reconciler.go

```diff
@@ -126,20 +148,39 @@ func (i *Identity) DeleteWorkloadIdentity(ctx context.Context, projectID string,
 	}
 	if kServiceAccount != nil && len(kServiceAccount.OwnerReferences) == 1 {
 		logging.FromContext(ctx).Desugar().Debug("Removing iam policy binding.")
-		if err := i.removeIamPolicyBinding(ctx, projectID, identifiable.IdentitySpec().GoogleServiceAccount, kServiceAccount); err != nil {
+		if err := i.removeIamPolicyBinding(ctx, projectID, googleServiceAccount, kServiceAccount); err != nil {
 			status.MarkWorkloadIdentityFailed(identifiable.ConditionSet(), deleteWorkloadIdentityFailed, err.Error())
 			return fmt.Errorf("removing iam policy binding failed with: %w", err)
 		}
 	}
 	return nil
 }
 
-func (i *Identity) createServiceAccount(ctx context.Context, namespace, gServiceAccount, clusterName string) (*corev1.ServiceAccount, error) {
-	kServiceAccountName := resources.GenerateServiceAccountName(gServiceAccount, clusterName)
+// getGoogleServiceAccountName will return Google service account name and corresponding raw Kubernetes service account name.
+func (i *Identity) getGoogleServiceAccountName(ctx context.Context, identifiable duck.Identifiable, clusterName string) (string, string, error) {
```

Comment:
My preference would be to have this return a struct with all the strings we will need in the various functions:

```golang
type wiNames struct {
  KSA string
  GSA string
  Namespace string
  ClusterName string
}
```